packages/makeup-hoist-element/index.js

```diff
@@ -0,0 +1,79 @@
+"use strict";
+
+var modalEl;
+var modalPlaceholder;
+var containerDiv;
+var prevAriaHiddenSetting;
+var bodyIndexes = [];
+
+function unhoist() {
+  if (modalEl) {
+    if (containerDiv) {
+      var childList = Array.from(containerDiv.childNodes);
+      childList.forEach(function (child) {
+        if (child.src === undefined) {
+          var index = bodyIndexes.shift();
+
+          if (index > document.body.childNodes.length) {
+            document.body.appendChild(child);
+          } else {
+            document.body.insertBefore(child, document.body.childNodes[index + 1]);
+          }
+        }
+      });
+      containerDiv.remove();
+      containerDiv = null;
+      bodyIndexes = [];
+    }
+
+    if (modalPlaceholder) {
+      modalEl.removeAttribute('aria-hidden'); // eslint-disable-next-line eqeqeq
+
+      if (prevAriaHiddenSetting != null) {
+        modalEl.setAttribute('aria-hidden', prevAriaHiddenSetting);
+      }
+
+      prevAriaHiddenSetting = null;
+      modalPlaceholder.replaceWith(modalEl); // let observers know the element is unhoisted
+
+      var event = document.createEvent('Event');
+      event.initEvent('unhoist', true, true);
+      modalEl.dispatchEvent(event);
+      modalPlaceholder = null;
+    }
+
+    modalEl = null;
+  }
+
+  return modalEl;
+}
+
+function hoist(el) {
+  unhoist();
```

Comment:
I would actually change this to check if the element is hoisted or not and then if it is not, then don't do anything.